File: day6/day6.kt
import java.io.File

fun main() {
    val input = File("day6.txt").readText().removeSuffix("\n").split("\n\n")
    val groups = input.map { t -> t.split("\n").map { s -> s.toSet() } }
    println(groups.map { g -> g.reduce { a, b -> a union b }.size }.sum())
    println(groups.map { g -> g.reduce { a, b -> a intersect b }.size }.sum())
}

File: day6/day6.py
data = open('day6.txt').read().rstrip().split('\n\n')
groups = [s.split('\n') for s in data]
count_any = lambda g: len(set(''.join(g)))
count_all = lambda g: len(set.intersection(*map(set, g)))
print(sum(map(count_any, groups)))
print(sum(map(count_all, groups)))
